buildSrc/src/main/kotlin/ScalaPublishingConventionsPlugin.kt

```diff
@@ -0,0 +1,75 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.BasePlugin
+import org.gradle.api.plugins.BasePluginExtension
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.register
+import org.gradle.plugins.signing.SigningExtension
+
+class ScalaPublishingConventionsPlugin : Plugin<Project> {
+  override fun apply(project: Project): Unit = project.run {
+    val scaladocJarTask: TaskProvider<Jar> = tasks.register<Jar>("scaladocJar") {
+      group = BasePlugin.BUILD_GROUP
+      tasks.findByName("scaladoc")?.let { dependsOn(it) }
```

Comment:
What if `scaladoc` is not available? 🤔 Does this task still do anything?

Alternatively, we can only add `scaladocJar` if `scaladoc` is available or I would use `requireNotNull` with a meaningful message here. (I am not entirely sure if we should throw `IllegalStateExeption` in Gradle though, I think we need to use the proper reporting tools).

Reply:
Good point. I guess we should control all the required settings. For instance, the signing key is mandatory when publishing to Maven Central. Let me think about it 🤔